akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbStore.scala

```diff
@@ -42,9 +42,24 @@ private[persistence] trait LeveldbStore extends Actor with WriteJournalBase with
 
   def asyncWriteMessages(messages: immutable.Seq[AtomicWrite]): Future[immutable.Seq[Try[Unit]]] =
     Future.fromTry(Try {
-      withBatch(batch ⇒ messages.map { a ⇒
-        Try(a.payload.foreach(message ⇒ addToMessageBatch(message, batch)))
-      })
+      withBatch { batch ⇒
+        var result: immutable.Seq[Try[Unit]] = Nil
+        val iter = messages.iterator
+        var n = 0
+        while (iter.hasNext) {
+          try {
+            iter.next().payload.foreach(message ⇒ addToMessageBatch(message, batch))
+            n += 1
+          } catch {
+            case NonFatal(e) ⇒
+              if (result.isEmpty)
+                result = Vector.fill(n)(AsyncWriteJournal.successUnit)
+              result :+= Failure(e)
+          }
+        }
+
+        result
+      }
```

Comment:
here is the second run, with the change here reverted:

    [info] PersistentActorThroughputBenchmark.actor_normalActor_reply_baseline                         thrpt  200  143879.774 ± 2125.070  ops/s
    [info] PersistentActorThroughputBenchmark.persistentActor_noPersist_reply                          thrpt  200  150487.166 ± 2621.352  ops/s
    [info] PersistentActorThroughputBenchmark.persistentActor_persistAsync_reply                       thrpt  200  111451.952 ± 2536.184  ops/s
    [info] PersistentActorThroughputBenchmark.persistentActor_persistAsync_replyRightOnCommandReceive  thrpt  200  114634.707 ± 2604.804  ops/s
    [info] PersistentActorThroughputBenchmark.persistentActor_persist_reply                            thrpt  200    9686.627 ±   36.865  ops/s 

i.e. no difference